fuel-crypto/src/signature.rs

```diff
@@ -12,49 +12,28 @@ use core::{fmt, str};
 pub struct Signature(Bytes64);
 
 impl Signature {
-    /// Memory length of the type
+    /// Memory length of the type in bytes.
     pub const LEN: usize = Bytes64::LEN;
 
-    /// Add a conversion from arbitrary slices into owned
+    /// Construct a `Signature` directly from its bytes.
     ///
-    /// # Safety
-    ///
-    /// There is no guarantee the provided bytes will be a valid signature. Internally, some FFI
-    /// calls to `secp256k1` are performed and we might have undefined behavior in case the bytes
-    /// are not canonically encoded to a valid `secp256k1` signature.
-    pub unsafe fn from_bytes_unchecked(bytes: [u8; Self::LEN]) -> Self {
+    /// This constructor expects the given bytes to be a valid signature. No signing is performed.
+    pub fn from_bytes(bytes: [u8; Self::LEN]) -> Self {
```

Comment:
What happens if this is not a valid signature? I know we use this function in fuel core with bytes from the network so maybe we need to add `from_bytes_checked -> Option<Self>`

Reply:
Afaik there's no way to know if the signature is valid until we actually try to validate the transaction, as we don't know what data has been signed at this point.

It should be fine to use this with bytes read from the network as if the signature isn't valid, the transaction validation will fail because the signature won't be valid for some spend or whatever.